physical/dynamodb/dynamodb.go

```diff
@@ -497,15 +500,39 @@ func (d *DynamoDBBackend) HAEnabled() bool {
 func (d *DynamoDBBackend) batchWriteRequests(requests []*dynamodb.WriteRequest) error {
 	for len(requests) > 0 {
 		batchSize := int(math.Min(float64(len(requests)), 25))
-		batch := requests[:batchSize]
+		batch := map[string][]*dynamodb.WriteRequest{ d.table: requests[:batchSize] }
 		requests = requests[batchSize:]
 
+		var err error
+
 		d.permitPool.Acquire()
-		_, err := d.client.BatchWriteItem(&dynamodb.BatchWriteItemInput{
-			RequestItems: map[string][]*dynamodb.WriteRequest{
-				d.table: batch,
-			},
-		})
+
+		boff := backoff.NewExponentialBackOff()
+		boff.MaxElapsedTime = 600 * time.Second
+
+		for len(batch) > 0 {
+			output, err := d.client.BatchWriteItem(&dynamodb.BatchWriteItemInput{
+				RequestItems: batch,
+			})
+
+			if err != nil{
+				break
```

Comment:
i don't think this is setting the outside-of-loop `err` to the inside-of-loop `err`, was that your intention?

I was curious about this so I spun up a quick example, and I believe it validates that the `err` on line 506 would remain `nil` in this case

example: https://play.golang.org/p/94zycqbFglQ

thoughts?